### PR TITLE
Docs - Add document for connector

### PR DESCRIPTION
**Pull request by huyleminh01**
## Summary
Refactor js module document:
- Add connector and core pages
- Add plugin npm2yarn
- Config to show more level on TOC

localhost_3000_js-module_connector

## Issue
(issue link here)

## Lasting Changes (Technical)

(please list down: code changes/things that have wide-effect; new libraries/functions added that can be used by others; examples below)

* (Added `class EmailValidator` to validate email address' validity)
* (Added `Tenant#is_trial?` check)

## Checklist

Please check directly on the box once each of these are done

- [ ] Documentation (if necessary)
- [ ] Tests (integration test/unit test)
- [ ] Integration Tests Passed
- [ ] Code Review

### CODE REVIEW

File: dbml-homepage/docs/js-module/connector.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Supported Databases
+
+- PostgreSQL
+- MySQL
+- MSSQL
+
+## Prerequisites
+
+Before you begin, ensure you have met the following requirements:
```
Comment (huyphung1602) [inline dbml-homepage/docs/js-module/connector.md:17]: I think:
We already have **Prerequisites** header. So I think you could shorten this one to:

```markdown
Please ensure you have met the following requirements:
...
```

File: dbml-homepage/docs/js-module/connector.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+- **Node.js**: You need to have Node.js version 18.0.0 or higher installed.
```
Comment (huyphung1602) [inline dbml-homepage/docs/js-module/connector.md:19]: There is only one requirement her. I suggest:

```
## Prerequisites

- Require Nodejs version 18.0.0 or higher.
```

File: dbml-homepage/configs/sidebars.ts [hunk 3]
```diff
@@ -12,7 +12,29 @@ import { SidebarsConfig } from '@docusaurus/plugin-content-docs';
...
 const SidebarConfigs: SidebarsConfig = {
-  docs: ['home', 'docs', 'cli', 'js-module'],
+  docs: [
+    'home',
+    'docs',
+    'cli',
+    {
+      type: 'category',
+      label: 'JS Module',
+      collapsed: true,
```
Comment (nguyenalter) [inline dbml-homepage/configs/sidebars.ts:22]: Why don't we expand the JS Module content by default? The current sidebar has many spaces so I prefer to expand it.

File: dbml-homepage/docs/js-module/connector.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+const { connector } = require('@dbml/connector');
+
+const connection = 'postgresql://dbml_user:dbml_pass@localhost:5432/schema';
+const format = 'postgres';
+
+const schemaJson = await connector.fetchSchemaJson(connection, format);
+```
+
+:::info
+You can find the definition of the `DatabaseSchema` [on this page](https://github.com/holistics/dbml/blob/a4dcb110f1d79f5d95b0d3db4b919914439e039d/packages/dbml-connector/src/connectors/types.ts#L89).
```
Comment (nguyenalter) [inline dbml-homepage/docs/js-module/connector.md:55]: "find the definition" looks ambiguous. I suggest changing this sentence to:
- The type definition of `DatabaseSchema` object can be found [here](https://github.com/holistics/dbml/blob/a4dcb110f1d79f5d95b0d3db4b919914439e039d/packages/dbml-connector/src/connectors/types.ts#L89).

File: dbml-homepage/docusaurus.config.ts [hunk 5]
```diff
@@ -28,8 +29,11 @@ const config: Config = {
       {
         docs: {
           sidebarPath: './configs/sidebars.ts',
-          breadcrumbs: false,
+          breadcrumbs: true,
```
Comment (nguyenalter) [inline dbml-homepage/docusaurus.config.ts:32]: Why do we need to update this prop?

File: dbml-homepage/docs/js-module/connector.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+npm install @dbml/connector
+```
+
+## APIs
+
+```javascript
+const { connector } = require('@dbml/connector');
+```
+
+### `fetchSchemaJson(connection, format)`
```
Comment (nguyenalter) [inline dbml-homepage/docs/js-module/connector.md:33]: I see that the `@dbml/core` has different writing styles for these named exports (e.g. `importer`).
image

Please move `fetchSchemaJson` to a sub-section of `connector` to ensure the writing styles are consistent.

**Review by nguyenalter — COMMENTED** [review]
Please help check if you can disable `reviewdog` check for hardcoded connection string in markdown files.

File: dbml-homepage/docusaurus.config.ts (hunk 5 above)
Comment (huyleminh01) [reply dbml-homepage/docusaurus.config.ts:32]: I just enabled this bar
Screenshot 2024-08-19 at 16 00 55

File: dbml-homepage/docs/js-module/connector.md (hunk 6 above)
Comment (huyleminh01) [reply dbml-homepage/docs/js-module/connector.md]: Fixed in commit https://github.com/holistics/dbml/pull/615/commits/790438dd7ef00ab26cbd219a7ffe50634ddc5923

File: dbml-homepage/docs/js-module/connector.md (hunk 4 above)
Comment (huyleminh01) [reply dbml-homepage/docs/js-module/connector.md]: Fixed in commit https://github.com/holistics/dbml/pull/615/commits/790438dd7ef00ab26cbd219a7ffe50634ddc5923

File: dbml-homepage/configs/sidebars.ts (hunk 3 above)
Comment (huyleminh01) [reply dbml-homepage/configs/sidebars.ts]: Fixed in commit https://github.com/holistics/dbml/pull/615/commits/790438dd7ef00ab26cbd219a7ffe50634ddc5923

File: dbml-homepage/docs/js-module/connector.md (hunk 2 above)
Comment (huyleminh01) [reply dbml-homepage/docs/js-module/connector.md]: Fixed in commit https://github.com/holistics/dbml/pull/615/commits/790438dd7ef00ab26cbd219a7ffe50634ddc5923

**Review by nguyenalter — APPROVED** [review]

**Review by huyphung1602 — APPROVED** [review]

**Pull request merged**